.github/workflows/validations.yaml

```diff
@@ -89,6 +89,61 @@ jobs:
           key: snapshot-build-${{ github.run_id }}
 
 
+  Upload-Snapshot-Artifacts:
+    # Note: changing this job name requires making the same update in the .github/workflows/release.yaml pipeline
+    name: "Upload snapshot artifacts"
+    needs: [Build-Snapshot-Artifacts]
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #v4.2.2
+
+      - name: Bootstrap environment
+        uses: ./.github/actions/bootstrap
+        with:
+          download-test-fixture-cache: true
+
+      - name: Download snapshot build
+        id: snapshot-cache
+        uses: actions/cache/restore@d4323d4df104b026a6aa633fdb11d772146be0bf #v4.2.2
+        with:
+          path: |
+            snapshot
+            .task
+          fail-on-cache-miss: true
+          key: snapshot-build-${{ github.run_id }}
+
+      # workaround for https://github.com/actions/cache/issues/1265
+      - name: (cache-miss) Snapshot build missing
+        if: steps.snapshot-cache.outputs.cache-hit != 'true'
+        run: echo "unable to download snapshots from previous job" && false
+
+      - run: npm install @actions/artifact@2.2.2
+
+      - uses: actions/github-script@v7
```

Comment:
why not use https://github.com/actions/upload-artifact directly as an action?

Reply:
@wagoodman the problem using upload-artifact is that you are only able to upload a single artifact -- if you give it 4 files, for example, it combines them into a single zip and uploads that. I would rather have these available for download individually so an interested party doesn't have to download _all_ builds just to get the windows build, for example